features/Add-Note-context-menu/Add-Note-context-menu.js

```diff
@@ -0,0 +1,299 @@
+                <p>${note.content.substring(0, maxCharLength)}${note.content.length > maxCharLength ? '...' : ''}</p>
+                <footer class="note__footer">
+                    <button id="editDetails-${index}" class="note-content__footer-button note__footer-button">
+                        <img src="/public/images/svg/page-edit.svg" alt="edit note" />
+                    </button>
+                    <button id="deleteDetails-${index}" class="note-content__footer-button note__footer-button">
+                        <img src="/public/images/svg/bin.svg" alt="delete note" />
+                    </button>
+                </footer>
+                `;
+            // Store the original index to use when viewing details
+            noteElement.dataset.index = notes.length - 1 - index;
+            noteElement.addEventListener('click', () => viewNoteDetails(noteElement.dataset.index));
+
+            const editButton = noteElement.querySelector(`#editDetails-${index}`);
+            const deleteButton = noteElement.querySelector(`#deleteDetails-${index}`);
+
+
+            // Attach the note actions
+            noteContextUpdateDeleteButton(editButton, deleteButton, noteElement.dataset.index);
+
+            displayCardsSection.appendChild(noteElement);
+        });
+    };
+
+    //****************************Note context*************************** */
+    function viewNoteDetails(index) {
+        const note = notes[index];
+        const noteDetails = document.createElement('dialog');
+        noteDetails.id = 'note-details';
+        noteDetails.innerHTML = `
+           <header class="note__header">
+                <h2>${note.title}</h2>
+                <button id="closeDetails" class="button-details">&times;</button>
+            </header>
+            <p class="note__content">${note.content}</p>
+            <footer class="note__footer">
+                <button id="editDetails" class="note-content__footer-button">
+                    <img src="/public/images/svg/page-edit.svg" alt="edit note" />
+                </button>
+                <button id="deleteDetails" class="note-content__footer-button">
+                    <img src="/public/images/svg/bin.svg" alt="delete note" />
+                </button>
+            </footer>
+        `;
+
+        document.body.appendChild(noteDetails);
+        noteDetails.showModal(); // Display the modal
+
+        document.getElementById('closeDetails').addEventListener('click', () => {
+            noteDetails.close(); // Close the modal
+            noteDetails.remove(); // Remove from DOM
+        });
+
+
+        const editButton = document.getElementById('editDetails');
+        const deleteButton = document.getElementById('deleteDetails');
+
+        // Attach the note actions
+        noteContextUpdateDeleteButton(editButton, deleteButton, index, noteDetails);
+    };
+
+    function noteContextUpdateDeleteButton(editButton, deleteButton, index, noteDetails = null) {
```

Comment:
Use onclick where we have actions on buttons, we do not need stop propagation if the 'click' is not generic.